### DISCUSSION TITLE
Cycle history of prefix

### DISCUSSION THREAD

**Discussion by justanotheranonymoususer** [chrisant996/clink#358] [Q&A] [answered]
Hi, thanks for Clink, it's great!

I'm using zsh with zsh-autosuggestions on my Mac and Linux, and one features that I'm using a lot is typing a prefix and then cycling through history with this prefix. Clink has the autosuggestion feature, but it ignores the prefix when I start to cycle.

For example, I use the following commands:

```
cd C:\one
cd C:\two
echo hello
<many other commands>
echo world
```

Now, I want to run `cd C:\one` again. If I type "cd ", I get a suggestion for `cd C:\two`, and I expect to click on the up arrow and see `cd C:\one`. Instead, I see `echo world` and all other unrelated commands.

An alternative is using Ctrl+R but I don't like it that much. If there's no solution in Clink, perhaps I'll have to get used to it.

Please let me know if there's any solution already in Clink and if not, whether it can be added. I can open a new issue if necessary.

**Comment by pukkandan** [comment, marked as answer]
See https://chrisant996.github.io/clink/clink.html#commands-for-manipulating-the-history

What you want is `history-search-backward` and forward. You can bind them to up/down in your inputrc file. Ref: https://chrisant996.github.io/clink/clink.html#keybindings

**Reply by justanotheranonymoususer** [reply]
Thank you!
